src/main/java/com/epam/test_generator/web_security/WebSecurityConfig.java

```diff
@@ -0,0 +1,79 @@
+package com.epam.test_generator.web_security;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.builders.WebSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.config.http.SessionCreationPolicy;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter;
+
+
+@EnableWebSecurity
+@Configuration
+
+public class WebSecurityConfig extends WebSecurityConfigurerAdapter {
+
+
+    private final JwtAuthenticationProvider authenticationProvider;
+    private final JwtAuthenticationEntryPoint jwtAuthenticationEntryPoint;
+
+    @Autowired
+    public WebSecurityConfig(JwtAuthenticationProvider authenticationProvider,
+        JwtAuthenticationEntryPoint jwtAuthenticationEntryPoint) {
+        this.authenticationProvider = authenticationProvider;
+        this.jwtAuthenticationEntryPoint = jwtAuthenticationEntryPoint;
+    }
+
+    @Bean
+    @Override
+    public AuthenticationManager authenticationManagerBean() throws Exception {
+        return super.authenticationManagerBean();
+    }
+
+    @Bean
+    JwtAuthenticationFilter jwtAuthenticationFilter() throws Exception {
+        JwtAuthenticationFilter jwtAuthenticationFilter = new JwtAuthenticationFilter();
+        jwtAuthenticationFilter.setAuthenticationManager(authenticationManagerBean());
+        jwtAuthenticationFilter
+            .setAuthenticationSuccessHandler(new JwtAuthenticationSuccessHandler());
+        jwtAuthenticationFilter.setAuthenticationFailureHandler(
+            new JwtAuthenticationFailureHandler());
+        return jwtAuthenticationFilter;
+}
+
+
+    @Bean
+    public PasswordEncoder encoder() {
+        return new BCryptPasswordEncoder();
+    }
+
+
+    @Override
+    protected void configure(HttpSecurity http) throws Exception {
```

Comment:
Provide detailed explanations of what this method does and add examples of how to modify code in case of 1 developers add controller that should be protected by jwt auth, 2 developer adds controller that should not be protected by jwt 3 developer adds controller and wants to permit access to it only for limited set of registered users.

Reply:
There is no any config now for controllers.
Now you can just exclude page from authentication, there is no any roles.